File: erpnext/hr/report/supervicion_almacen/supervicion_almacen.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

# import frappe

import requests
from requests.adapters import HTTPAdapter, Retry

def execute(filters=None):
	columns, data = [], []
	session = requests.Session()
	retry = Retry(connect=3, backoff_factor=0.5)
	adapter = HTTPAdapter(max_retries=retry)
	session.mount('http://', adapter)
	session.mount('https://', adapter)
	supervisor = session.post("https://recursoshumanos.shalom.com.pe/api/supervicion-almacen-reporte", json = filters)
	supervisor = supervisor.json()

	if "data" in supervisor:
		return get_columns() , supervisor.get("data")

def get_columns():
	columns = [
		{
			'label': "ZONA NACIONAL",
			'fieldname': 'zona_nacional',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "SUPERVISOR",
			'fieldname': 'supervisor',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "AGENCIA",
			'fieldname': 'branch',
			'fieldtype': 'Data',
			'width': 300 ,
		},
		{
			'label': "TIPO DE AGENCIA",
			'fieldname': 'tipo_agencia',
			'fieldtype': 'Data',
			'width': 300 ,
		},
		{
			'label': "TIPO DE ALMACEN",
			'fieldname': 'tipo_almacen',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "CATEGORIA",
			'fieldname': 'categoria',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "CREADO POR",
			'fieldname': 'nombre_completo',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "ESTADO",
			'fieldname': 'status',
			'fieldtype': 'Data',
			'width': 300
		},
		{
			'label': "CHECK",
			'fieldname': 'check',
			'fieldtype': 'Check',
			'width': 300
		}

	]

	return columns

File: erpnext/hr/report/supervicion_almacen/supervicion_almacen.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */
let today = frappe.datetime.get_today();
frappe.query_reports["Supervicion Almacen"] = {
	"filters": [
		{
			"fieldname":"date_init",
			"label": __("From"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_start(frappe.datetime.get_today()),
			"width": "100px"
		},
		{
			"fieldname":"date_end",
			"label": __("Hasta"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_end(frappe.datetime.get_today()),
			"width": "100px"
		},
		{
			"fieldname":"branch",
			"label": __("Branch"),
			"fieldtype": "Link",
			"options": "Branch",
			"width": "100px",
		},
		{
			"fieldname":"zona_nacional",
			"label": __("Zonas Nacional"),
			"fieldtype": "Link",
			"options": "Zonas Nacional",
			"width": "100px",
		},
		{
			"fieldname":"tipo_almacen",
			"label": __("Warehouse Type"),
			"fieldtype": "Link",
			"options": "Warehouse Type",
			"width": "100px",
		},
	]
};

